be/src/storage/rowset/segment_iterator.cpp

```diff
@@ -1577,6 +1577,9 @@ Status SegmentIterator::_do_get_next(Chunk* result, vector<rowid_t>* rowid) {
                 continue;
             }
             ColumnPtr& col = chunk->get_column_by_index(column_index++);
+            if (!_scan_range.empty()) {
+                RETURN_IF_ERROR(_column_iterators[cid]->seek_to_ordinal(_scan_range.begin()));
```

Comment:
in what situation this seek to ordinal is expected?

Reply:
`[array_column_iterator.cpp:365] Check failed: range.empty() || (range.begin() == _array_size_iterator->get_current_ordinal())`
This indicates that when executing the get_io_range_vec function, there is a case where the range begin is not equal to the current ordinal of _array_size_iterator. Therefore, seek_to_ordinal is added here. @starrocks-xupeng